tests/model_fields/test_jsonfield.py

```diff
@@ -636,6 +636,43 @@ def test_has_key_number(self):
                     [obj],
                 )
 
+    def test_has_key_special_chars(self):
+        value = {
+            "single'": [],
+            'double"': [
+                connection.features.supports_double_quotes_in_json_field_lookups,
+            ],
```

Comment:
I structured the tests in this way because before I tried the custom delimiter for Oracle, I couldn't get it to work. So, I opted to add another flag for single quotes specifically.

If we are sure that the custom delimiter approach is "bulletproof", then we can probably simplify this test. However, if we find a breaking test case with the custom delimiter, then we might need another flag after all.